tincr/io/library/genCellLibrary.tcl

```diff
@@ -879,10 +987,25 @@ proc ::tincr::create_xml_cell_library { {part xc7a100t-csg324-3} {filename ""} {
         puts $xml_out "    </cell>"
         remove_cell $cell_instance -quiet
     }
-
+    
     create_port_xml $site_map $xml_out
     
     puts $xml_out "  </cells>"
+    
+    # Create the xml for each macro cell supported in the device
+    puts "Creating macro definitions..."
+    set supported_macros [tincr::get_supported_macro_cells]
+    
+    if {[llength $supported_macros] > 0} {
+        puts $xml_out "  <macros>"
+        foreach macro_lib_cell $supported_macros {
+            set macro_cell [create_cell -reference $macro_lib_cell "tmp" -quiet]
+            tincr::write_macro_xml $macro_cell $xml_out
+            remove_cell $macro_cell -quiet
```

Comment:
This function is in the cell library generator tcl script. It should not be called with an open project, and I don't think it would worth the time opening a new project for each cell I add to the cell library XML. I suppose we could add a check to make sure that no design is currently opened when the cell library generator is called.